dev/NavigationView/NavigationView.cpp

```diff
@@ -1594,6 +1734,222 @@ void NavigationView::UpdateVisualStateForDisplayModeGroup(const winrt::Navigatio
     }
 }
 
+void NavigationView::OnNavigationViewItemTapped(const winrt::IInspectable& sender, const winrt::TappedRoutedEventArgs& args)
+{
+    if (auto nvi = sender.try_as<winrt::NavigationViewItem>())
+    {
+        if (IsSettingsItem(nvi))
+        {
+            OnSettingsInvoked();
+        }
+        else
+        {
+            OnNavigationViewItemInvoked(nvi);
+        }
+        nvi.Focus(winrt::FocusState::Pointer);
+    }
+}
+
+void NavigationView::OnNavigationViewItemKeyUp(const winrt::IInspectable& sender, const winrt::KeyRoutedEventArgs& args)
+{
+    // Because ListViewItem eats the events, we only get these keys on KeyUp.
+    if (args.OriginalKey() == winrt::VirtualKey::GamepadA)
+    {
+        if (auto nvi = sender.try_as<winrt::NavigationViewItem>())
+        {
+            HandleKeyEventForNavigationViewItem(nvi, args);
+        }
+    }
+}
+
+void NavigationView::OnNavigationViewItemKeyDown(const winrt::IInspectable& sender, const winrt::KeyRoutedEventArgs& args)
+{
+    if (auto nvi = sender.try_as<winrt::NavigationViewItem>())
+    {
+        HandleKeyEventForNavigationViewItem(nvi, args);
+    }
+}
+
+void NavigationView::HandleKeyEventForNavigationViewItem(const winrt::NavigationViewItem& nvi, const winrt::KeyRoutedEventArgs& args)
+{
+    auto key = args.Key();
+    if (IsSettingsItem(nvi))
+    {
+        // Because ListViewItem eats the events, we only get these keys on KeyDown.
```

Comment:
>// Because ListViewItem eats the events, we only get these keys on KeyDown. [](start = 8, length = 75)

There is a preview key up we could use instead if that is a better design.

Reply:
See my response in https://github.com/microsoft/microsoft-ui-xaml/pull/1683#discussion_r364018648